tests/juniper/juniper_base_protocol_test.py

```diff
@@ -469,6 +469,168 @@ def test_interface_trunk_native_vlan_merge(self):
         }))
         assert_that(result.xpath("data/configuration/vlans/vlan"), has_length(0))
 
+    def test_interface_set_trunk_native_vlan_then_set_members_after(self):
+        self.edit({
+            "vlans": [
+                {"vlan": [
+                    {"name": "VLAN2995"},
+                    {"vlan-id": "2995"}]},
+                {"vlan": [
+                    {"name": "VLAN2996"},
+                    {"vlan-id": "2996"}]},
+                {"vlan": [
+                    {"name": "VLAN2997"},
+                    {"vlan-id": "2997"}]},
+            ],
+            "interfaces": {
+                "interface": [
+                    {"name": "ge-0/0/3"},
+                    {"unit": [
+                        {"name": "0"},
+                        {"family": {
+                            "ethernet-switching": {
+                                self.PORT_MODE_TAG: "trunk"
+                            }}}]}]}})
+        self.nc.commit()
+
+        self.edit({
+            "interfaces": {
+                "interface": [
+                    {"name": "ge-0/0/3"},
+                    {"unit": [
+                        {"name": "0"},
+                        {"family": {
+                            "ethernet-switching": {
+                                "native-vlan-id": "2995"
+                            }}}]}]}})
+        self.nc.commit()
+
+        self.edit({
+            "interfaces": {
+                "interface": [
+                    {"name": "ge-0/0/3"},
+                    {"unit": [
+                        {"name": "0"},
+                        {"family": {
+                            "ethernet-switching": {
+                                "vlan": [
+                                    {"members": "2997"},
+                                ]}}}]}]}})
+        self.nc.commit()
+
+        result = self.nc.get_config(source="running", filter=dict_2_etree({"filter": {
+            "configuration": {"interfaces": {"interface": {"name": "ge-0/0/3"}}}}
+        }))
+
+        assert_that(result.xpath("data/configuration/interfaces/interface"), has_length(1))
+
+        int003 = result.xpath("data/configuration/interfaces/interface")[0]
+        assert_that(int003.xpath("unit/family/ethernet-switching/native-vlan-id")[0].text, equal_to("2995"))
+
+        self.cleanup(vlan("VLAN2995"), vlan("VLAN2996"), vlan("VLAN2997"),
+                     interface("ge-0/0/3", [self.PORT_MODE_TAG, "vlan", "native-vlan-id"]))
+        result = self.nc.get_config(source="running", filter=dict_2_etree({"filter": {
+            "configuration": {"vlans": {}}}
+        }))
+        assert_that(result.xpath("data/configuration/vlans/vlan"), has_length(0))
```

Comment:
I am going to go under the assumption, that this might have to be done because of what is in memory due to the nature of how we are testing the fake switches. So this might be more functional than not. And best to test that the cleanup was done properly before the next test.